packages/features/bookings/lib/handleSeats.ts

```diff
@@ -0,0 +1,796 @@
+  invitee: Invitee;
+  allCredentials: Awaited<ReturnType<typeof getAllCredentials>>;
+  organizerUser: OrganizerUser;
+  originalRescheduledBooking: OriginalRescheduledBooking;
+  bookerEmail: string;
+  tAttendees: TFunction;
+  bookingSeat: BookingSeat;
+  reqUserId: number | undefined;
+  rescheduleReason: RescheduleReason;
+  reqBodyUser: string | string[] | undefined;
+  noEmail: NoEmail;
+  isConfirmedByDefault: IsConfirmedByDefault;
+  additionalNotes: AdditionalNotes;
+  reqAppsStatus: ReqAppsStatus;
+  attendeeLanguage: string | null;
+  paymentAppData: PaymentAppData;
+  fullName: ReturnType<typeof getFullName>;
+  smsReminderNumber: SmsReminderNumber;
+  eventTypeInfo: EventTypeInfo;
+  uid: short.SUUID;
+  eventTypeId: EventTypeId;
+  reqBodyMetadata: ReqBodyMetadata;
+  subscriberOptions: GetSubscriberOptions;
+  eventTrigger: WebhookTriggerEvents;
+}) => {
+  const loggerWithEventDetails = createLoggerWithEventDetails(eventType.id, reqBodyUser, eventType.slug);
+
+  let resultBooking:
+    | (Partial<Booking> & {
+        appsStatus?: AppsStatus[];
+        seatReferenceUid?: string;
+        paymentUid?: string;
+        message?: string;
+        paymentId?: number;
+      })
+    | null = null;
+
+  const booking = await prisma.booking.findFirst({
+    where: {
+      OR: [
+        {
+          uid: rescheduleUid || reqBookingUid,
+        },
+        {
+          eventTypeId: eventType.id,
+          startTime: evt.startTime,
+        },
+      ],
+      status: BookingStatus.ACCEPTED,
+    },
+    select: {
+      uid: true,
+      id: true,
+      attendees: { include: { bookingSeat: true } },
+      userId: true,
+      references: true,
+      startTime: true,
+      user: true,
+      status: true,
+      smsReminderNumber: true,
+      endTime: true,
+      scheduledJobs: true,
+    },
+  });
+
+  if (!booking) {
+    throw new HttpError({ statusCode: 404, message: "Could not find booking" });
+  }
+
+  // See if attendee is already signed up for timeslot
+  if (
+    booking.attendees.find((attendee) => attendee.email === invitee[0].email) &&
+    dayjs.utc(booking.startTime).format() === evt.startTime
+  ) {
+    throw new HttpError({ statusCode: 409, message: "Already signed up for this booking." });
+  }
+
+  // There are two paths here, reschedule a booking with seats and booking seats without reschedule
+  if (rescheduleUid) {
+    // See if the new date has a booking already
+    const newTimeSlotBooking = await prisma.booking.findFirst({
+      where: {
+        startTime: evt.startTime,
+        eventTypeId: eventType.id,
+        status: BookingStatus.ACCEPTED,
+      },
+      select: {
+        id: true,
+        uid: true,
+        attendees: {
+          include: {
+            bookingSeat: true,
+          },
+        },
+      },
+    });
+
+    const credentials = await refreshCredentials(allCredentials);
+    const eventManager = new EventManager({ ...organizerUser, credentials });
```

Comment:
newTimeSlotBooking should be a findFirstOrThrow and these two constants should be called after.